src/subcommand.rs

```diff
@@ -255,10 +255,22 @@ impl Subcommand {
     Ok(())
   }
 
-  fn format(config: &Config, ast: Ast, search: &Search) -> Result<(), Error<'static>> {
+  fn format(config: &Config, ast: Ast, src: &str, search: &Search) -> Result<(), Error<'static>> {
     config.require_unstable("The `--fmt` command is currently unstable.")?;
 
-    if let Err(io_error) =
+    if config.dry_run {
+      let new_src = ast.to_string();
+      if src != new_src {
```

Comment:
```suggestion
      if src != src_formatted {
```

Reply:
I pulled out `src_formatted` to outer scope too